x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -613,6 +622,19 @@ public void onResponse(Void unused) {
         }
     }
 
+    private IndicesAccessControl wrapIndicesAccessControl(IndicesAccessControl indicesAccessControl) {
+        if (indicesAccessControl == null
+            || indicesAccessControl == IndicesAccessControl.ALLOW_NO_INDICES
+            || indicesAccessControl == IndicesAccessControl.DENIED
+            || indicesAccessControl == IndicesAccessControl.allowAll()
+            || indicesAccessControl.getClass().equals(IndicesAccessControl.class) == false) {
+            // Wrap only if it's not one of the statically defined nor already wrapped.
+            return indicesAccessControl;
+        } else {
+            return indicesAccessControlWrapper.wrap(indicesAccessControl);
+        }
+    }
```

Comment:
I think it would be easier to follow if this method is a static one inside the DlsFlsFeatureTrackingIndicesAccessControlWrapper class.